site-src/geps/gep-724.md

```diff
@@ -0,0 +1,624 @@
+  namespace: foo
+  labels:
+    gateway: xlb
+spec:
+  rules:
+  - name: abc
+    matches:
+    - path: /bar
+---
+kind: ReferencePolicy
+metadata:
+  name: infra-gateways
+  namespace: foo
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    namespace: infra
+  to:
+  - group: gateway.networking.k8s.io
+    kind: HTTPRoute
+```
+
+#### Advantages
+
+* Consistent use of ReferencePolicy throughout the API.
+* Provides a single way of binding Gateways to Routes.
+
+#### Disadvantages
+
+* Even the simplest cross-namespace reference from Gateway -> Route would
+  require a ReferencePolicy in each target namespace.
+* Existing demos and examples would become significantly more verbose.
+* Does not suppport attaching Routes to arbitrary parents.
+* Does not prevent accidental exposure of Routes.
+* Route owners have limited control in terms of which Gateways their Route is
+  attached to.
+
+### 2. ReferencePolicy with Routes referencing Gateways
+
+![ReferencePolicy with Routes referencing Gateways](images/724-alt2.png)
+
+The other way we could use ReferencePolicy would be with Routes referencing
+Gateways. Unfortunately the nested structure of Gateways makes this nearly
+impossible to do effectively. A core concept for Gateways is that each listener
+should be able to attach to an entirely different set of Routes. For example,
+a Gateway may want to delegate foo.com to the foo namespace and bar.com to the
+bar namespace. Unfortunately that would be very difficult to recreate with
+ReferencePolicy.
+
+ReferencePolicy is fundamentally about trusting references from resource of kind
+Foo in to resources of kind Bar. Names and section names are intentionally
+excluded. If we added both of those concepts to ReferencePolicy, this would be
+possible, but quite complex and verbose. This is what the example from above
+would look like with this approach:
+
+```yaml
+kind: Gateway
+metadata:
+  name: lb
+  namespace: infra
+spec:
+  listeners:
+  - name: foo
+    hostname: foo.com
+    port: 80
+---
+kind: ReferencePolicy
+metadata:
+  name: foo-lb
+  namespace: infra
+spec:
+  from:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+  to:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: lb
+    sectionName: foo
+---
+kind: HTTPRoute
+metadata:
+  name: foo
+  namespace: foo
+spec:
+  parentRefs:
+  - kind: Gateway
+    namespace: infra
+    name: lb
+    sectionName: foo
+  rules:
+  - name: abc
+    matches:
+    - path: /bar
+```
+
+#### Advantages
+
+* Consistent use of ReferencePolicy throughout the API.
+* Provides a single way of binding Gateways to Routes.
+* Supports attaching Routes to arbitrary parents.
+* Prevents accidental exposure of Routes.
+
+#### Disadvantages
+
+* In most cases, each listener in a Gateway would require a unique
+  ReferencePolicy resource.
+* Even the simplest cross-namespace reference from Route -> Gateway would
+  require a ReferencePolicy in each target namespace.
```

Comment:
Maybe worth calling out explicitly that this point would rule out self-service use-cases (if I understand correctly).  Most likely, I, as an application developer, do not have permission to create a ReferencePolicy in the infra namespace.

Reply:
Added, thanks!